tuf/data/types.go

```diff
@@ -127,6 +128,68 @@ type FileMeta struct {
 	Custom json.RawMessage `json:"custom,omitempty"`
 }
 
+// CheckHashes verifies all the checksums specified by the "hashes" of the payload.
+func CheckHashes(payload []byte, hashes Hashes) error {
+	// k, v indicate the hash algorithm and the corresponding value
+	for k, v := range hashes {
+		switch k {
+		case notary.SHA256:
+			checksum := sha256.Sum256(payload)
+			if subtle.ConstantTimeCompare(checksum[:], v) == 0 {
+				return fmt.Errorf("%s checksum mismatched", k)
+			}
+		case notary.SHA512:
+			checksum := sha512.Sum512(payload)
+			if subtle.ConstantTimeCompare(checksum[:], v) == 0 {
+				return fmt.Errorf("%s checksum mismatched", k)
+			}
+		default:
+			return fmt.Errorf("unsupported hash algorithm")
+		}
+	}
+
+	return nil
```

Comment:
This would return true if there were no hashes for the given target.  Could we add a check here to make sure at least one supported checksum hash is included in the `Hashes` object?